Березин Максим: проектная работа №1 - первый pull request

MBerezin opened the pull request.

MBerezin committed:
add new class - Calculator

MBerezin committed:
add people counter logic

MBerezin committed:
add constructor

MBerezin committed:
Revert "add constructor"

This reverts commit 0a10f99a04a46b9548887a87ca16424270f0838a.

MBerezin committed:
add constructor for Calculator

MBerezin committed:
add new class - Product

MBerezin committed:
add constructor for Product

MBerezin committed:
add new get-methods

MBerezin committed:
add addProduct-method

MBerezin committed:
add "add products"-logic

MBerezin committed:
change decimal delimiter to "."

MBerezin committed:
decompose logic into methods

MBerezin committed:
add new methods to show calculation result

MBerezin committed:
set default locale
add new methode to show division result

MBerezin committed:
remove not used import

MBerezin committed:
add check for negative price

gusar-off commented on src/main/java/Main.java:
```diff
@@ -1,8 +1,99 @@
+        Locale.setDefault(Locale.US);
+
+        Scanner scanner = new Scanner(System.in);
+        int peopleCounter;
```
Можно объединить инициализацию переменной с ее объявлением
```
int peopleCounter = countPeople(scanner);
Calculator calculator = new Calculator(peopleCounter);
```

gusar-off commented on src/main/java/Main.java:
```diff
@@ -1,8 +1,99 @@
+        double productPrice;
+        String productName;
+        Product product;
+
```
Допускается один отступ, а здесь 3

MBerezin replied:
Между 4-ой и 8-ой строкой?

gusar-off commented on src/main/java/Main.java:
```diff
@@ -1,8 +1,99 @@
+
+            productName = scanner.next();
+
+            if (productName.equalsIgnoreCase("ЗАВЕРШИТЬ")){
```
"ЗАВЕРШИТЬ" лучше вынести в статическую константу

gusar-off commented on src/main/java/Calculator.java:
```diff
@@ -0,0 +1,43 @@
+    }
+
+    private String getCurrencyWriting(double price){
+        int roundedPrice = (int)Math.floor(price);
```
`Math.floor` округляет до целого в меньшую сторону, а приведение к `int` отбрасывает дробную часть. То есть это по сути эквивалентные операции, но с разными типами в результате. Поэтому нет смысла делать и то и другое. Лучше сделать только `Math.floor` потому что это более явно.

gusar-off commented on src/main/java/Calculator.java:
```diff
@@ -0,0 +1,43 @@
+        return String.format("%,.2f", price);
+    }
+
+    private String getCurrencyWriting(double price){
```
Немного некорректно рассчитывается окончание. Получается если цена будет 121 рубль, метод вернет "рублей". Правильный алгоритм звучит так:
Проверить лежит ли остаток от деления на 100 в интервале от 11 до 14 включительно, если да, то возвращаем "рублей", если нет, то берем остаток от деления на 10 и прогоняем с твоими проверками.
То есть нужно во-первых проверять именно остаток от деления, во-вторых учесть, что числа заканчивающиеся на 11-12-13-14 и заканчивающиеся на 1-2-3-4, будут иметь разные окончания

MBerezin committed:
fix logic for currency writing

MBerezin committed:
- add constant for ending of products input
- delete redundant indents
- unite declaration and initialisation some variables
